enemy/enemy.gd

```diff
@@ -55,10 +55,16 @@ func _handle_hit(_amount: int) -> void:
 	add_child(particle)
 
 func _enter_tree() -> void:
+	SignalBus.enemy_died.connect(_handle_enemy_died)
 	hit.connect(_handle_hit)
+	add_to_group("enemy")
 	
 func _on_area_2d_area_entered(_area: Area2D) -> void:
 	fsm.change_state("attack")
 
 func _on_area_2d_area_exited(_area: Area2D) -> void:
 	fsm.change_state("idle")
+	
+func _handle_enemy_died(enemy: Enemy, _reward: int) -> void:
+	if enemy == self:
+		remove_from_group("enemy")
```

Comment:
If the `enemy` on the signal is for this, then we might not need it, I included a `died` signal, not in the bus but in the enemy itself, so we should be able to connect to it and it's only called per-instance, so it would be ideal for this.